modules/nativescript/nativescript.cpp

```diff
@@ -754,7 +784,16 @@ NativeScriptInstance::~NativeScriptInstance() {
 	script_data->destroy_func.destroy_func((godot_object *)owner, script_data->destroy_func.method_data, userdata);
 
 	if (owner) {
+
+#ifndef NO_THREADS
+		script->owners_lock->lock();
+#endif
+
 		script->instance_owners.erase(owner);
+
+#ifndef NO_THREADS
+		script->owners_lock->lock();
+#endif
```

Comment:
There also seems to be unprotected access to NativeScriptLanguage in NativeScript::get_script_desc.